identity-core/src/crypto/merkle_key/verifier.rs

```diff
@@ -132,41 +143,69 @@ where
 
   // Validate the signature algorithm tag
   if tag_s != S::TAG {
-    return Err(Error::InvalidMerkleSignatureKeyTag(Some(tag_s)));
+    return Err(MerkleSignatureKeyTagError(Some(tag_s)).into());
   }
 
   // Validate the digest algorithm tag
   if tag_d != D::TAG {
-    return Err(Error::InvalidMerkleDigestKeyTag(Some(tag_d)));
+    return Err(MerkleDigestKeyTagError(Some(tag_d)).into());
   }
 
   // Extract and return the Merkle root hash
   key
     .merkle_key
     .get(2..)
     .and_then(Hash::from_slice)
-    .ok_or(Error::InvalidKeyFormat)
+    .ok_or_else(|| VerificationError::ProcessingFailed("invalid key format".into()))
 }
 
-fn expand_signature_value(signature: &SignatureValue) -> Result<(PublicKey, Vec<u8>, Vec<u8>)> {
+/* If this function gets used outside of verification as well, we might have to come up with a better error type.
+Since this function is private this is not a problem in terms of stability.
+*/
+fn expand_signature_value(signature: &SignatureValue) -> Result<(PublicKey, Vec<u8>, Vec<u8>), VerificationError> {
   let data: &str = signature.as_str();
   let mut parts: _ = data.split('.');
 
   // Split the signature data into `public-key/proof/signature`
-  let public: &str = parts.next().ok_or(Error::InvalidProofFormat)?;
-  let proof: &str = parts.next().ok_or(Error::InvalidProofFormat)?;
-  let signature: &str = parts.next().ok_or(Error::InvalidProofFormat)?;
+  let public: &str = parts.next().ok_or(InvalidProofFormat)?;
+  let proof: &str = parts.next().ok_or(errors::InvalidProofFormat)?;
+  let signature: &str = parts.next().ok_or(errors::InvalidProofFormat)?;
 
   // Extract bytes of the base58-encoded public key
-  let public: PublicKey = decode_b58(public)
-    .map_err(|_| Error::InvalidProofFormat)
+  let public: PublicKey = utils::decode_b58(public)
+    .map_err(|_| errors::InvalidProofFormat)
```

Comment:
This could be made more consistent by removing the `errors::` prefix in a couple of places.